### PR TITLE
feat: support custom optimizers in logistic regression

### PR DESCRIPTION
**Pull request by polvalente**

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -32,8 +32,9 @@ defmodule Scholar.MixProject do
       {:ex_doc, "~> 0.29", only: :docs},
       {:nx, github: "elixir-nx/nx", sparse: "nx", override: true},
       {:nimble_options, "~> 0.5.2 or ~> 1.0"},
-      {:exla, ">= 0.0.0", only: [:test, :dev]},
-      {:explorer, "~> 0.5.1", only: [:test, :dev]}
+      {:exla, github: "elixir-nx/exla", sparse: "exla", only: [:test, :dev]},
```
Comment (josevalim) [inline mix.exs:35]: Why this change? :)

**Review by josevalim — APPROVED** [review]

File: mix.exs (hunk 1 above)
Comment (polvalente) [reply mix.exs:35]: I was getting some weird C++ raise error that went away with this update ¯\_(ツ)_/¯
